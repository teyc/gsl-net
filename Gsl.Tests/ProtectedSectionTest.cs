using System.IO;
using Xunit;
using Xunit.Abstractions;

namespace Gsl.Tests
{
    public class ProtectedSectionTest : TestBase
    {
        public ProtectedSectionTest(ITestOutputHelper log) : base(log)
        {
        }

        [Fact]
        public void ProtectedSectionsShouldPreserveUserContent()
        {
            const string template = "data/protected.gsl";
            const string data = "data/protected.json";

            _fileSystem.File.WriteAllText("Customer.cs", File.ReadAllText(Path.DataFile("data/Customer.cs")));
            TemplateWithSingleFileOutput(template, data);
        }
    }
}
