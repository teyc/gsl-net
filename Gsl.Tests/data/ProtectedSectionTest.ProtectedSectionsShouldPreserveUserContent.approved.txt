namespace CodeGen.Demo
{
  class Customer
  {
    public string FirstName 
    { 
        get 
        {
			// CUSTOM-CODE-BEGIN get-FirstName
            return _firstName;
			// CUSTOM-CODE-END get-FirstName
        } 
    }
    public string LastName 
    { 
        get 
        {
			// CUSTOM-CODE-BEGIN get-LastName
			return _lastName;
			// CUSTOM-CODE-END get-LastName
        } 
    }
    public string DateOfBirth 
    { 
        get 
        {
			// CUSTOM-CODE-BEGIN get-DateOfBirth

			// CUSTOM-CODE-END get-DateOfBirth
        } 
    }
  }
}
}